pkgs/native_assets_cli/lib/src/config.dart

```diff
@@ -404,6 +406,168 @@ class BuildOutputBuilder extends HookOutputBuilder {
       syntax.BuildOutput.fromJson(super._syntax.json);
 }
 
+extension DataAssetsDirectoryExtension on BuildOutputBuilder {
+  /// Extension on [BuildOutput] to handle data asset directories and files.
+  ///
+  /// This extension provides a convenient way for build hooks to add
+  /// [DataAsset] dependencies from one or more directories or individual files.
+  ///
+  /// If any specified path does not exist, a [FileSystemException] is thrown.
+  /// Any error during the directory listing is caught and rethrown with
+  /// additional context.
+  Future<void> addDataAssetDirectories(
+    List<String> paths, {
+    required BuildInput input,
+  }) async {
+    final packageName = input.packageName;
+    final packageRoot = input.packageRoot;
+    for (final path in paths) {
+      final resolvedUri = packageRoot.resolve('$packageName/$path');
+      final directory = Directory.fromUri(resolvedUri);
+      final file = File.fromUri(resolvedUri);
+
+      if (await directory.exists()) {
+        // Process as a directory.
+        addDependency(directory.uri);
+        try {
+          await for (final entity in directory.list(
+            recursive: true,
+            followLinks: false,
+          )) {
+            // Add dependency for every file and directory found.
+            addDependency(entity.uri);
+          }
+        } on FileSystemException catch (e) {
+          throw FileSystemException(
+            'Error reading directory "$path": ${e.message}',
+            directory.path,
+            e.osError,
+          );
+        }
+      } else if (await file.exists()) {
+        addDependency(file.parent.uri);
```

Comment:
I don't believe this is needed. If we specify a file by itself, then simply that file as dependency should be enough.

Reply:
Test `addDataAssetDirectories processes one file` fails if I remove this, so it does not show up in output.dependencies. Is there something upstream that adds the folder path?

Reply:
Cool, this should match that flow now.